include/camp/array.hpp

```diff
@@ -155,6 +155,12 @@ namespace camp {
          }
       }
 
+      CAMP_HOST_DEVICE constexpr void swap(array& a) noexcept(is_nothrow_swappable<T>::value) {
+         for (std::size_t i = 0; i < N; ++i) {
+            safe_swap(elements[i], a[i]);
+         }
```

Comment:
Given the conditional noexcept specification, this should probably be exception safe right?

Reply:
Should I make safe_swap conditionally noexcept if the type is move constructible and move assignable without throwing?

Reply:
I just updated safe_swap. Let me know if that's what you had in mind.

Reply:
That would work, but if you want this to provide the strong exception guarantee when the arrays contain a potentially throwing move and/or copy, it will need a temporary array to use the copy and swap pattern.  Otherwise an exception can leave both arrays in an inconsistent state.  If the move constructor is noexcept then there's no problem, so it might even make sense to make that a requirement for this method to pass sfinae.

Reply:
I made the swap function conditional on the move constructor/assignment operator being noexcept.